lib/bootsnap/load_path_cache/path_scanner.rb

```diff
@@ -20,20 +23,23 @@ def self.call(path)
         # and the bundle path is '.bundle'.
         contains_bundle_path = BUNDLE_PATH.start_with?(path)
 
-        dirs = []
-        requirables = []
+        dirs = Set.new
```

Comment:
I have mixed feelings on this. On one hand, 

> [`Set#to_a` Converts the set to an array. The order of elements is uncertain.](https://ruby-doc.org/stdlib-2.4.1/libdoc/set/rdoc/Set.html#method-i-to_a)

...but on the other hand, 

> [Hashes enumerate their values in the order that the corresponding keys were inserted.](https://ruby-doc.org/core-2.4.1/Hash.html)

...and...

> [Set uses Hash as storage, ...](https://ruby-doc.org/stdlib-2.4.1/libdoc/set/rdoc/Set.html)

So while it's relatively clear that this is in fact safe, I don't like that it relies on behaviour that is explicitly undefined.

Reply:
Order of either dirs or requirables is irrelevant. The only place where it is used is [Cache](https://github.com/Shopify/bootsnap/blob/master/lib/bootsnap/load_path_cache/cache.rb#L147-L148) where they are transformed to hash. The uncertainty of the order of `Set#to_a` is even more irrelevant if you know that order of entries yielded from `glob` is uncertain as well.

Let me paste benchmark comparing current implementation with the previous one:
```
require 'bootsnap'
require 'benchmark/ips'

Benchmark.ips do |b|
  b.report('after') { Bootsnap::LoadPathCache::PathScanner.call($LOAD_PATH.last) }
  b.report('before') { Bootsnap::LoadPathCache::PathScanner.call($LOAD_PATH.last) }

  b.hold! __FILE__.sub('.rb', '.json')
  b.compare!
end

# Warming up --------------------------------------
#               before    15.000  i/100ms
# Calculating -------------------------------------
#               before    153.094  (± 3.9%) i/s -    765.000  in   5.005778s
# 
# Comparison:
#                after:      484.5 i/s
#               before:      153.1 i/s - 3.16x  slower
```

Reply:
Scratch that, I have found better glob pattern that does not yield duplicates, sets are not necessary anymore.